corehq/apps/accounting/models.py

```diff
@@ -1702,6 +1729,31 @@ def exists_for_domain(cls, domain):
     def get_domain(self):
         return self.subscription.subscriber.domain
 
+    @classmethod
+    def autopayable_invoices(cls, date_due):
+        """ Invoices that can be auto paid on date_due """
+        invoices = (cls.objects.
+                    select_related('subscription__account').
+                    filter(subscription__account__auto_pay_user__isnull=False).
+                    filter(date_due=date_due))
+        return invoices
+
+    def pay_invoice(self, amount, payment_record):
+        """ Creates a CreditAdjustment (thru adding credits) and updates the invoice balance """
+        CreditLine.add_credit(
```

Comment:
i think i've seen this pattern in another PR where we add and then subtract the amount. i agree it's necessary, but think we should encapsulate this because anyone who isn't familiar with the billing code is going to scratch their head